app/controllers/sp_components_controller.rb

```diff
@@ -57,6 +58,27 @@ def destroy
     redirect_to admin_path(anchor: component&.component_type)
   end
 
+  def associate_service
+    component_present = SpComponent.exists?(params[:sp_component_id])
+    service = Service.find_by_id(params[:service_id])
+
+    if component_present && service.present?
+      @event = AssignSpComponentToServiceEvent.create(service: Service.find_by_id(params[:service_id]), sp_component_id: params[:sp_component_id])
```

Comment:
Why not? reduce the calls by Service.find_by_id and change Service.exists? on top to Service.find...
```
    component = SpComponent.find_by_id(params[:sp_component_id])
    service = Service.find_by_id(params[:service_id])

    if component.present? && service.present?
      @event = AssignSpComponentToServiceEvent.create(service: service, sp_component_id: component.id)

      if @event.valid?
        redirect_to sp_components_path
      else
        Rails.logger.info(@event.errors.full_messages)

        render :show
      end
    else
      flash[:error] = I18n.t('service.errors.unknown_component') unless component.present?
```